cli/command/cli.go

```diff
@@ -209,12 +209,13 @@ func (cli *DockerCli) Initialize(opts *cliflags.ClientOptions, ops ...Initialize
 
 	cli.configFile = cliconfig.LoadDefaultConfigFile(cli.err)
 
+	cli.contextStore = store.New(cliconfig.ContextStoreDir(), cli.contextStoreConfig)
+	cli.currentContext, err = resolveContextName(opts.Common, cli.configFile, cli.contextStore)
+	if err != nil {
+		return err
+	}
+
 	if cli.client == nil {
-		cli.contextStore = store.New(cliconfig.ContextStoreDir(), cli.contextStoreConfig)
-		cli.currentContext, err = resolveContextName(opts.Common, cli.configFile, cli.contextStore)
-		if err != nil {
-			return err
-		}
 		endpoint, err := resolveDockerEndpoint(cli.contextStore, cli.currentContext, opts.Common)
```

Comment:
This is, below the context here, used as `cli.dockerEndpoint = endpoint`. Should that be pulled out too?

Reply:
I wasn't sure but yep you are right, let's move it up too 👍